src/ResourceManager/Websites/Commands.Websites/Models.WebApp/WebAppOptionalSlotBaseCmdlet.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Management.Automation;
+using Microsoft.Azure.Commands.WebApps.Models;
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using Microsoft.Azure.Management.WebSites.Models;
+
+namespace Microsoft.Azure.Commands.WebApps
+{
+    // For some cmdlets, the slot is optional, but will be used if specified.
+    public class WebAppOptionalSlotBaseCmdlet : WebAppBaseClientCmdLet
+    {
+        protected const string ParameterSet1Name = "S1";
```

Comment:
Pnemonic names for parameter sets are better - basically I think your two parameter sets are something like FromWebApp and FromResource